projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php

```diff
@@ -61,3 +61,35 @@ function robots_txt( string $output, $public ): string {
 }
 
 add_filter( 'robots_txt', __NAMESPACE__ . '\\robots_txt', 12, 2 );
+
+/**
+ * Disable the Open Graph Tags based on the value of either wpcom_public_coming_soon option.
+ */
+function remove_og_tags() {
+	if ( ! (bool) get_option( 'wpcom_public_coming_soon' ) ) {
+		return;
+	}
+
+	// Disable Jetpack Open Graph Tags.
+	if ( function_exists( 'jetpack_og_tags' ) ) {
+		// @phan-suppress-next-line PhanUndeclaredFunctionInCallable
+		remove_action( 'wp_head', 'jetpack_og_tags' );
+	}
+
+	// Avoid calling check_open_graph as it registers the jetpack_og_tags function when running wp_head action.
+	// @phan-suppress-next-line PhanUndeclaredClassInCallable
+	if ( class_exists( '\Jetpack', false ) && is_callable( 'Jetpack::init' ) ) {
+		// @phan-suppress-next-line PhanUndeclaredClassMethod
+		$jetpack = \Jetpack::init();
+		remove_action( 'wp_head', array( $jetpack, 'check_open_graph' ), 1 );
+	}
```

Comment:
Would it be an option to simplify this logic and just force the Open Graph off with the `jetpack_enable_open_graph` filter?

I haven't tested but I would expect something like this to work (maybe with some priority changes?)

```suggestion
	add_filter( 'jetpack_enable_open_graph', '__return_false' );
```

If that works, I wouldn't be surprised if Yoast SEO offered a similar filter allowing one to shortcircuit their implementation.

Reply:
The `check_open_graph` function would [register the hook to turn on the open graph if certain modules are active](https://github.com/Automattic/jetpack/blob/4c74a9d0047e36d85e69278bbd0518651ead2e5b/projects/plugins/jetpack/class.jetpack.php#L1972) so I'm not sure whether it's okay to do that.

However, after re-reading the code, the function also adds the following filter to turn off the open graph if there is any conflict plugin. So, I think I can simply add the same filter.
```php
add_filter( 'jetpack_enable_open_graph', '__return_false', 99 );
```

> If that works, I wouldn't be surprised if Yoast SEO offered a similar filter allowing one to shortcircuit their implementation.

The current way is recommended by https://developer.yoast.com/customization/yoast-seo/disabling-yoast-seo/. We can simply remove all actions for `wpseo_head` but I think it would be better to follow the official suggestion. Does it make sense?

Reply:
> but I think it would be better to follow the official suggestion

Yeah, definitely!